apps/dashboard/src/components/activity/activity-table.tsx

```diff
@@ -0,0 +1,172 @@
+  const { currentEnvironment } = useEnvironment();
+  const hoverTimerRef = useRef<NodeJS.Timeout | null>(null);
+  const [searchParams] = useSearchParams();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const { activities, isLoading, hasMore } = useFetchActivities({ filters });
+
+  const page = parseInt(searchParams.get('page') || '0');
+
+  const handlePageChange = (newPage: number) => {
+    const newParams = createSearchParams({
+      ...Object.fromEntries(searchParams),
+      page: newPage.toString(),
+    });
+    navigate(`${location.pathname}?${newParams}`);
+  };
+
+  const handleRowMouseEnter = (activity: IActivity) => {
+    hoverTimerRef.current = setTimeout(() => {
+      queryClient.prefetchQuery({
+        queryKey: [QueryKeys.fetchActivity, currentEnvironment?._id, activity._id],
+        queryFn: () => getNotification(activity._id, currentEnvironment!),
+      });
+    }, 1000);
+  };
+
+  const handleRowMouseLeave = () => {
+    if (hoverTimerRef.current) {
+      clearTimeout(hoverTimerRef.current);
+      hoverTimerRef.current = null;
+    }
+  };
+
+  useEffect(() => {
+    return () => {
+      if (hoverTimerRef.current) {
+        clearTimeout(hoverTimerRef.current);
+      }
+    };
+  }, []);
+
+  return (
+    <div className="flex min-h-full min-w-[800px] flex-1 flex-col">
+      <Table
+        isLoading={isLoading}
+        loadingRow={<SkeletonRow />}
+        containerClassname="border-x-0 border-b-0 border-t border-t-neutral-200 rounded-none shadow-none"
+      >
+        <TableHeader>
+          <TableRow>
+            <TableHead className="h-9 px-3 py-0">Event</TableHead>
+            <TableHead className="h-9 px-3 py-0">Status</TableHead>
+            <TableHead className="h-9 px-3 py-0">Steps</TableHead>
+            <TableHead className="h-9 px-3 py-0">Triggered Date</TableHead>
+          </TableRow>
+        </TableHeader>
+        <TableBody>
+          {activities.map((activity) => (
+            <TableRow
+              key={activity._id}
+              className={cn(
+                'relative cursor-pointer hover:bg-neutral-50',
+                selectedActivityId === activity._id &&
+                  'bg-neutral-50 after:absolute after:right-0 after:top-0 after:h-[calc(100%-1px)] after:w-[5px] after:bg-neutral-200'
+              )}
+              onClick={() => onActivitySelect(activity)}
+              onMouseEnter={() => handleRowMouseEnter(activity)}
+              onMouseLeave={handleRowMouseLeave}
+            >
+              <TableCell className="px-3">
+                <div className="flex flex-col">
+                  <span className="text-foreground-950 font-medium">
+                    {activity.template?.name || 'Deleted workflow'}
+                  </span>
+                  <span className="text-foreground-400 text-[10px] leading-[14px]">
+                    {activity.transactionId} {getSubscriberDisplay(activity.subscriber)}
```

Comment:
I also wanted, the problem is that we don't have spacing for properly placing it on hover because of the subscriber details on the right side, and adding padding on the left made it look weird. @twentyone24 do you have any suggestions? we can make it copiable without the button just by hovering with a tooltip Click to Copy?

Reply:
I didn't get the problem, here is how it's in the workflows list:
![Screenshot 2024-12-12 at 09 52 57](https://github.com/user-attachments/assets/5c4c5e5a-e0f9-48b5-b222-d6439972cb1c)

We can do the same in the activity table:
![Screenshot 2024-12-12 at 09 53 24](https://github.com/user-attachments/assets/35a26515-0ead-4ab6-89a8-89c936f09908)